### DISCUSSION TITLE
PROFILING A GOPRO'S LINEAR GAMMA FOR POST

### DISCUSSION THREAD

**Discussion by IRCGraphic** [gopro/labs#450] [Q&A] [answered]
I've tried without success to profile a GoPro's linear gamma for working in post, the way I usually do it is by capturing raws and video samples with 1 f-stop increments of a neutral color surface illuminated by light and then with Davinci Resolve set the raws to linear gamma and manually calibrate each f-stop of the unknown video samples gamma to match its linear equivalent, save it as a 1D lut and use it for converting any GoPro gamma to a Davinci Resolve standard for advanced video editing. 
My question is I haven't find any Labs code to match the same exact ISO/shutter speed settings for both raw and video, is there a way to do it? Or any source where I can find the exact parameters for a Flat gamma curve for each the GP9 and GP11?

Thanks

**Comment by dnewman-gpsw** [comment]
The "Linear gamma" seems to be is mixed terminology, as linear light is gamma 1.0.  So linear gamma is weird.

Protune Flat is log, it has the transfer function:    output = log(input*112+1)/log(113)  (input and output are 0.0 to 1.0)

Protune Flat is the only non-locally tone-mapped mode on the camera, all others modes can't be reversed, as LTM and GTM (global tone-mapping) are applied, and both are image dependent.  So your testing will get confusing results.

see https://www.desmos.com/calculator/sfxqtunzty  For info on the Protune Flat Log curve 

The Protune curve is used in tools like https://cameramanben.github.io/LUTCalc/LUTCalc/index.html  (although LUT Calc assumes -1 EV when using Protune Flat.)

**Comment by IRCGraphic** [comment]
Thanks i will check Lutcalc, (linear gamma as the photometric light captured by the sensor) so i'm correct understanding that any GoPro models apply the same "Flat" log gamma over a linear 1.0 gamma?
About the transfer function formula and log designer, colour math profane here, with that formula and the custom ones of the log designer the video's gamma it's possible to virtually create log to linear 1.0 conversions? 
The f-stops method with physical samples is empirical, any possible video gamma ("non-locally tone-mapped") can be profiled to its raw linear gamma equivalent (in Davinci Resolve the gamma applied to raw dng's has a simple photometric management without any highlights recovery or anything weird that alters the ratios of light or exposure, with the help of a waveform and its tools to measure the levels an accurate enough manual profiling can be done between video and raw samples)

**Reply by dnewman-gpsw** [reply, marked as answer]
Yes, the transfer function is the direct mapping from linear light to the encoding curve. So the math is reversible, and you don't need to measure, just use the function provided.

**Reply by IRCGraphic** [reply]
Great, thanks!

**Reply by IRCGraphic** [reply]
Following the math provided I've created a public Github repository with 1D LUTs for converting Flat to Linear and viceversa available to download: https://github.com/IRCGraphic/GoPro-Flat-Gamma-Conversions/tree/main 

![](https://user-images.githubusercontent.com/113941057/254634402-edb0c5da-9168-4edb-874d-95478f475fa2.png)

**Reply by dnewman-gpsw** [reply]
handy

**Reply by IRCGraphic** [reply]
Ok I just found a way to match ISO/shutter speed settings between RAW and video (locking the min/max video shutter speed code) and comparing side by side I must say the RAW is just hair darker, what could be causing the discrepancy? I know for a fact that Davinci Resolve is consistent with the logarithmic scaling of exposure (each f-stop doubles or halves the gain intensity values) in linear gamma no matter the RAW stills camera source.
Also when setting the LOGB= to 1 it should cancel out any log curve and effectively result in a linear 1.0 gamma right? But it doesn't match either the reverse of your Flat gamma transfer function, LOGB=1 it's like almost 2 f-stops brighter. By the way it would be really cool to record in a true linear gamma for some creative and technical purposes.

**Reply by IRCGraphic** [reply]
Forgot to mention: when comparing a GoPro raw vs video with the same exposure (ISO/SS) and default lens setting (wide) I see there's some discrepancy also in the fisheye distortion?
And almost seems there's a vignetting correction in the video? Could the gamma discrepancy be caused by a vignetting correction under the hood?

**Reply by dnewman-gpsw** [reply]
You can't use RAW Photo to model video behavior, they have very different sensor readouts, likely resulting in differing black levels. 

Video is a more developed image including, lens flat-fielding and some lens distortion modifications.

There does seem to be a bug in the LOGB=113 curve applied as this is some addition pre-emphasis, I will dig into that when I get some time.

**Reply by IRCGraphic** [reply]
Thanks for your support, awesome work you're doing in Labs!
